Stack/Opc.Ua.Core/Types/Encoders/IComplexTypeInstance.cs

```diff
@@ -25,4 +27,17 @@ public interface IComplexTypeInstance
         /// </value>
         ExpandedNodeId TypeId { get; set; }
     }
+
+    /// <summary>
+    /// Interface implemented by complex data type instances that discover and provide type information dynamically
+    /// </summary>
+    public interface IDynamicComplexTypeInstance : IComplexTypeInstance
+    {
+        /// <summary>
+        /// Returns the xml qualified name for the type instance
+        /// </summary>
+        /// <returns>The xml qualified name for the type instance</returns>
+        XmlQualifiedName GetXmlName(IServiceMessageContext context);
+    }
```

Comment:
A IComplexTypeInstance implementation needs the Encodable Factory and the Namespace Table from the context. The reason is that the implementation may need to make additional nested types in a structure available via the Factory. Otherwise, all known types would have to be pre-registered, which largely defeats the purpose of a truly dynamic (= meta-data-driven) type implementation.

Reply:
@mregen Please let me know if you need any additional information or maybe want to discuss on a call?

Reply:
Hi @MarkusHorstmann , we discussed the PR yesterday in the user group and it is accepted, just taking a final check then its ready to go. Thanks!